convert_hf_to_gguf.py

```diff
@@ -3141,32 +3136,74 @@ def modify_tensors(self, data_torch: Tensor, name: str, bid: int | None) -> Iter
-        super().__init__(*args, **kwargs)
+    def __init__(self, dir_model: Path, ftype: gguf.LlamaFileType, fname_out: Path, is_big_endian: bool = False,
+                 use_temp_file: bool = False, eager: bool = False,
+                 metadata_override: Path | None = None, model_name: str | None = None,
+                 split_max_tensors: int = 0, split_max_size: int = 0, dry_run: bool = False,
+                 small_first_shard: bool = False, hparams: dict[str, Any] | None = None):
+
+        hparams = Model.load_hparams(dir_model)
+
+        if (hparams is None or hparams.get("moe_every_n_layers", 0) < 1):
+            self.model_arch = gguf.MODEL_ARCH.NOMIC_BERT
+        else:
+            self.model_arch = gguf.MODEL_ARCH.NOMIC_BERT_MOE
+
+        super().__init__(dir_model, ftype, fname_out, is_big_endian, use_temp_file,
+                         eager, metadata_override, model_name, split_max_tensors,
+                         split_max_size, dry_run, small_first_shard, hparams)
 
         # the HF config claims n_ctx=8192, but it uses RoPE scaling
         self.hparams["n_ctx"] = 2048
 
-        # SwigLU activation
-        assert self.hparams["activation_function"] == "swiglu"
+        if (self.hparams.get("moe_every_n_layers", 0) < 1):
+            assert self.hparams["activation_function"] == "swiglu"
+        else:
+            assert self.hparams["activation_function"] == "gelu"
+
         # this doesn't do anything in the HF version
         assert self.hparams["causal"] is False
-        # no bias tensors
-        assert self.hparams["qkv_proj_bias"] is False
-        assert self.hparams["mlp_fc1_bias"] is False
-        assert self.hparams["mlp_fc2_bias"] is False
+
+        if (self.hparams.get("moe_every_n_layers", 0) < 1):
+            assert self.hparams["qkv_proj_bias"] is False
+            assert self.hparams["mlp_fc1_bias"] is False
+            assert self.hparams["mlp_fc2_bias"] is False
+        else:
+            assert self.hparams["qkv_proj_bias"] is True
+            assert self.hparams["mlp_fc1_bias"] is True
+            assert self.hparams["mlp_fc2_bias"] is True
+
         # norm at end of layer
         assert self.hparams["prenorm"] is False
         # standard RoPE
         assert self.hparams["rotary_emb_fraction"] == 1.0
         assert self.hparams["rotary_emb_interleaved"] is False
         assert self.hparams["rotary_emb_scale_base"] is None
 
+    def modify_tensors(self, data_torch: torch.Tensor, name: str, bid: int | None) -> Iterable[tuple[str, torch.Tensor]]:
+        # If the tensor is an experts bias tensor, skip it by returning an empty list.
+        if "mlp.experts.bias" in name:
+            return []  # Explicitly return an empty list.
+
+        if "mlp.experts.mlp.w1" in name:
+            data_torch = data_torch.view(self.hparams["num_experts"], self.hparams["n_inner"], self.hparams["n_embd"])
+            return [(self.map_tensor_name(name) + ".weight", data_torch)]
```

Comment:
will this work? (no need to return here)

```suggestion
            name += ".weight"
```